pennylane/workflow/execution.py

```diff
@@ -267,7 +267,16 @@ def _make_inner_execute(
     """
 
     if isinstance(device, qml.devices.LegacyDevice):
-        device_execution = set_shots(device, override_shots)(device.batch_execute)
+        dev_execute = (
+            device.batch_execute
+            if execution_config is None
+            or not device.capabilities().get("supports_mid_measure", False)

```

Comment:
Added this condition because devices that don't support MCMs have `batch_execute` that only takes circuits as input. The other option is to update such devices' `batch_execute` to include a `**kwargs` in their function signature. However, that might still cause errors with plugin devices.